services/user/InviteSys/src/AuthInviteSys.cpp

```diff
@@ -0,0 +1,101 @@
+         return;  // Request is valid
+      else if (type == AuthInterface::runAsMatchedReq)
+         return;  // Request is valid
+      else if (type == AuthInterface::runAsMatchedExpandedReq)
+         abortMessage("runAs: caller attempted to expand powers");
+      else if (type == AuthInterface::runAsOtherReq)
+         abortMessage("runAs: caller is not authorized");
+      else if (type != AuthInterface::topActionReq)
+         abortMessage("unsupported auth type");
+
+      if (getInviteClaim(claims))
+      {
+         check(action.service == InviteSys::service, restrictedService.data());
+
+         // Compile time check that acceptCreate and reject are still methods in InviteSys
+         using T1 = decltype(&UserService::Invite::InviteSys::acceptCreate);
+         using T2 = decltype(&UserService::Invite::InviteSys::reject);
+         check(action.method == "acceptCreate"_m || action.method == "reject"_m,
+               restrictedActions.data());
+         return;
+      }
+
+      std::string err = "checkAuthSys: ";
+      err += missingInviteSig;
+      abortMessage(err);
+
+      // Billing notes: The point of an invite is to allow those without a psibase account to create one
+      //    in a way that doesn't suffer from the chicken/egg problem that you *already* need an account
+      //    to call an action (such as accepting an invite to create an account).
+      // Therefore, this service allows the invited-sys user to be used by those without an account,
+      //    specifically for calling the accept/reject invite actions on invite-sys. That means that if
+      //    it succeeds, invited-sys is the account that will be billed for the action. Invited-sys is a
+      //    system account and has infinite resources, so the billing will always succeed.
+      // This does not mean that no one paid for the action to accept an invite, because the action is
+      //    prepaid by the user who creates the invite.
+   }
+
+   void AuthInviteSys::canAuthUserSys(psibase::AccountNumber user)
+   {
+      check(user == InviteSys::payerAccount, notWhitelisted.data());
+   }
+
+   std::optional<Claim> AuthInviteSys::getInviteClaim(const std::vector<Claim>& claims)
+   {
+      for (const auto& claim : claims)
+      {
+         if (claim.service == SystemService::VerifyEcSys::service)
+         {
+            auto inviteKey = psio::convert_from_frac<PublicKey>(claim.rawData);
+            auto inviteOpt = Invite::InviteSys::Tables{Invite::InviteSys::service}
+                                 .open<Invite::InviteTable>()
+                                 .get(inviteKey);
+
+            if (inviteOpt.has_value())
+            {
+               return std::optional<Claim>{claim};
+            }
+         }
+      }
+      return std::nullopt;
+   }
+
+   void AuthInviteSys::requireAuth(const PublicKey& pubkey)
+   {
+      auto claimOpt = getInviteClaim(to<SystemService::TransactionSys>().getTransaction().claims);
```

Comment:
This doesn't need to look up the key in the invite table. Just checking the `{VerifyEcSys::service, pubkey}` is in `getTransaction().claims` is sufficient. Also, a user with an existing account can be prevented from calling `accept` if there is an active invite that uses his account's public key, because the invite lookup will find the wrong invitation.

Reply:
True - removed that part of the check